backend/app/models/campaign_charity.rb

```diff
@@ -4,6 +4,20 @@ class CampaignCharity < ApplicationRecord
   belongs_to :campaign
   belongs_to :charity
   has_many :secondary_donations, dependent: :destroy
+  has_many :coupons, through: :secondary_donations
 
   validates :giving_sg_url, presence: true, allow_blank: false, format: { with: URI::DEFAULT_PARSER.make_regexp }
+
+  def donation_breakdown
+    primary_donor_amount = coupons.count * campaign.coupon_denomination
+    secondary_donors_amount = secondary_donations.sum(&:amount)
+    total_amount = (primary_donor_amount + secondary_donors_amount).to_f
+    primary_donor_fraction = primary_donor_amount / total_amount
+    secondary_donors_fraction = secondary_donors_amount / total_amount
+
+    { primary_donor_amount: primary_donor_amount,
+      primary_donor_fraction: primary_donor_fraction,
+      secondary_donors_amount: secondary_donors_amount,
+      secondary_donors_fraction: secondary_donors_fraction }
+  end
```

Comment:
Why is the logic here repeated? 

Can we have this method in only one place? 

```suggestion
  def donation_breakdown
    campaign.donation_breakdown
  end
```

Reply:
It's not repeated, since campaign-level donation breakdown is different from campaignCharity-level donation breakdown.

Campaign level:
<img width="368" alt="image" src="https://user-images.githubusercontent.com/75519136/195785224-a96fe33a-4b47-4845-9c5b-d83e78b97a42.png">

CampaignCharity level:
<img width="365" alt="image" src="https://user-images.githubusercontent.com/75519136/195785273-77272da2-8092-4187-9b33-8cd75ee8c962.png">